doc/hardware/porting/board_porting.rst

```diff
@@ -242,6 +242,37 @@ followed by trial and error. If you want to understand details, you will need
 to read the rest of the devicetree documentation and the devicetree
 specification.
 
+Board extensions
+****************
+
+Board extensions are board fragments that may be present in any board root
+folder, under ``${BOARD_ROOT}/extensions/${BOARD}``. The board extension
+directory may contain Kconfig fragments and/or devicetree overlays. Board
+extensions are, by default, automatically loaded and applied on top of board
+files, before anything else. There is no guarantee on which order extensions are
+applied, in case multiple exist. This feature shall be disabled by passing
+``-DBOARD_EXTENSIONS=OFF`` when building.
+
+Board extensions are designed for downstream users, for example,
+``example-application`` or vendor SDKs. In some situations, certain hardware
+description or `choices <devicetree-chosen-nodes>`_ can not be added in the
+upstream Zephyr repository, but they can be in a downstream project, where
+custom bindings or driver classes can also be created. This feature may also be
+useful in development phases, when the board skeleton lives upstream, but other
+features are developed in a downstream module.
+
+Note that board extensions need to follow the
+:ref:`same guidelines <porting-general-recommendations>` as regular boards. For
+example, it is wrong to enable extra peripherals or subsystems in a board
+extension.
+
+.. warning::
+
+   Board extensions are not allowed in any module referenced in Zephyr's
+   ``west.yml`` manifest file. Any board changes are required to be submitted to
+   the main Zephyr repository. Exceptions may include standalone applications
+   included as modules such as bootloaders.
```

Comment:
Not sure on this, for things like mcuboot it already supports (and has) board specific overlays. If mcuboot had one of these added then it would be active throughout the whole zephyr tree, not just for mcuboot, so not sure I understand the point of allowing that

Reply:
The thing is that we can't forbid mcuboot to create extensions if they want (I doubt they need this feature, though). We could always build Mcuboot with BOARD_EXTENSIONS=OFF, though, assuming they design things properly so that a minimal build is available w/o extensions.

Reply:
If this is seen as too risky, we can put the requirement to any west.yml module, regardless of its type.

Reply:
Removed exception, I think it's better to prevent any cases for now.